app/assets/javascripts/annotations.js

```diff
@@ -0,0 +1,371 @@
+$(function() {
+
+  /* Highlights lines longer than 80 characters autolab red color w*/
+  var highlightLines = function(highlight) {
+    $(".code").each(function() {
+      var text = $(this).text();
+      // To account for lines that have 80 characters and a line break
+      var diff = text[text.length - 1] === "\n" ? 1 : 0;
+      if(text.length - diff > 80 && highlight){
+        $(this).css("background-color", "rgba(153, 0, 0, .9)");
+        $(this).prev().css("background-color", "rgba(153, 0, 0, .9)");
+      } else {
+        $(this).css("background-color", "white");
+        $(this).prev().css("background-color", "white");
+      }
+    });
+  };
+  $("#highlightLongLines").click(function(){
+    highlightLines(this.checked);
+  })
+
+  /* following paths/functions for annotations */
+  var path = '<%= course_assessment_submission_annotations_path(@course, @assessment, @submission) %>';
+  var createPath = function(annotation) {
+    return path + '.js';
+  };
+  var updatePath = function(annotation) {
+    return path + '/' + annotation.attr('id').replace(/[^0-9]*/, '') + '.js';
+  };
+  var deletePath = updatePath;
+
+  var problems = <%== @problems %>;
```

Comment:
Since rails does not pass the array of problems here. Perhaps we should move this into the page. What do you suggest?